CONTRIBUTING.md

```diff
@@ -0,0 +1,87 @@
+
+## Table of Contents
+
+- [Contributing Guidelines](#contributing-guidelines)
+  - [Table of Contents](#table-of-contents)
+  - [Community Guidelines](#community-guidelines)
+  - [Looking for ways to contribute?](#looking-for-ways-to-contribute)
+  - [Code Contribution Guidelines](#code-contribution-guidelines)
+    - [Discuss big changes as Issues first](#discuss-big-changes-as-issues-first)
+    - [Welcoming Pull Requests](#welcoming-pull-requests)
+    - [Code and Conventions](#code-and-conventions)
+      - [Commits](#commits)
+      - [Pull Requests](#pull-requests)
+      - [Tests](#tests)
+      - [Documentation](#documentation)
+      - [Writing Changelog Entries](#writing-changelog-entries)
+      - [Bumping Versions](#bumping-versions)
+      - [Mergify and the Send-it Label](#mergify-and-the-send-it-label)
+  - [Credits](#credits)
+
+## Community Guidelines
+
+At rust-libp2p, we value a friendly and inclusive community. Harassment and abusive behavior are strictly prohibited. Please be respectful and kind to fellow contributors.
+
+## Looking for ways to contribute?
+
+There are several ways you can contribute to rust-libp2p:
+- Start contributing immediately via the opened [issues](https://github.com/libp2p/rust-libp2p/issues).
+- Reporting issues, bugs, mistakes, or inconsistencies.
+- Submitting a feature request.
+- Opening a new pull request: primary means of making concrete changes to the code.
+
+## Code Contribution Guidelines
+
+### Discuss big changes as Issues first
+
+Before starting to code significant improvements, it's important to document them as GitHub issues in order to get feedback.
+
+### Welcoming Pull Requests
+
+Pull requests are the primary means of making concrete changes to the code, documentation, and dependencies of rust-libp2p. Even small pull requests, such as fixing a typo, are highly appreciated.
+
+### Code and Conventions
+
+#### Commits
+
+It is highly recommended to organize your changes into logically grouped commits. Keeping each commit focused on a specific aspect of the changes makes it easier for reviewers to understand and evaluate the modifications.
+
+Commit messages must start with a short subject line, followed by an optional, more detailed explanatory text which is separated from the summary by an empty line.
+
+#### Pull Requests
+
+To streamline the contribution process, we aim to simplify the PR template. Instead of including all the information in the template itself, we will provide clear guidelines and references to these new contributor guidelines. This approach helps maintain a concise and focused PR template while providing contributors with comprehensive instructions.
+
+When submitting pull requests, we strongly encourage squash merging. This means that we discourage force pushes, as it makes the diff between pushes easier for us to review. Squash merging allows us to maintain a clean and organized commit history.
```

Comment:
We don't just encourage squash-merging, we always squash-merge :)

We don't want force pushes because it messes with GitHub's review system.